components/camera/client.go

```diff
@@ -166,13 +166,15 @@ func (c *client) Properties(ctx context.Context) (Properties, error) {
 	if err != nil {
 		return Properties{}, err
 	}
-	result.IntrinsicParams = &transform.PinholeCameraIntrinsics{
-		Width:  int(resp.IntrinsicParameters.WidthPx),
-		Height: int(resp.IntrinsicParameters.HeightPx),
-		Fx:     resp.IntrinsicParameters.FocalXPx,
-		Fy:     resp.IntrinsicParameters.FocalYPx,
-		Ppx:    resp.IntrinsicParameters.CenterXPx,
-		Ppy:    resp.IntrinsicParameters.CenterYPx,
+	if intrinsics := resp.IntrinsicParameters; intrinsics != nil {
+		result.IntrinsicParams = &transform.PinholeCameraIntrinsics{
+			Width:  int(intrinsics.WidthPx),
+			Height: int(intrinsics.HeightPx),
+			Fx:     intrinsics.FocalXPx,
+			Fy:     intrinsics.FocalYPx,
+			Ppx:    intrinsics.CenterXPx,
+			Ppy:    intrinsics.CenterYPx,
+		}
```

Comment:
This was the bug fix for the original issue. Here, if `resp.IntrinsicParameters == nil` we would panic with the error
> invalid memory address or nil pointer dereference 

on line 171. 

Checking if the `resp.IntrinsicParameters == nil` before attempting to access its methods fixes the issue.